geode-wan/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ReplicateRegionCommandDUnitTest.java

```diff
@@ -210,90 +216,404 @@ public void testUnsuccessfulCancelReplicateRegionCommandInvocation_WithPartition
+        verifyResultOfStoppingParallelSender(result);
+      } else {
+        if (stopPrimarySender) {
+          verifyResultOfStoppingPrimarySerialSender(result);
+        } else {
+          verifyResultStoppingSecondarySerialSender(result);
+        }
+      }
 
-      CommandResultAssert replicateRegionCommand =
-          gfsh.executeAndAssertThat(command).statusIsSuccess();
-      replicateRegionCommand.hasTableSection().hasColumns().hasSize(3);
-      replicateRegionCommand.hasTableSection(ResultModel.MEMBER_STATUS_SECTION).hasColumn("Member")
-          .hasSize(3);
-      replicateRegionCommand.hasTableSection(ResultModel.MEMBER_STATUS_SECTION).hasColumn("Message")
-          .hasSize(3);
-      replicateRegionCommand.hasTableSection(ResultModel.MEMBER_STATUS_SECTION).hasColumn("Status")
-          .containsExactlyInAnyOrder("OK", "OK", "ERROR");
-      Condition<String> startsWithError = new Condition<String>(
-          s -> s.equals("Sender " + senderIdInA + " is not running"), "sender is not running");
-      Condition<String> haveEntriesReplicated =
-          new Condition<String>(s -> s.startsWith("Entries replicated:"), "entries replicated");
-      replicateRegionCommand.hasTableSection(ResultModel.MEMBER_STATUS_SECTION).hasColumn("Message")
-          .asList().haveExactly(1, startsWithError).haveExactly(2, haveEntriesReplicated);
 
-    } else {
-      serverInB.invoke(() -> stopReceivers());
+    } else if (gwToBeStopped == Gateway.RECEIVER) {
+      Thread.sleep(1500);
+      // Stop receiver
+      // if parallel sender: stop any receiver
+      // if serial sender: stop receiver connected to primary or secondary
+      if (useParallel) {
+        server2InB.invoke(() -> cache.close());
+      } else {
+        // Region type has no influence on which server should be stopped
+        if (stopPrimarySender) {
+          // Stop the first server which had an available receiver
+          server1InB.invoke(() -> cache.close());
+        } else {
+          server3InB.invoke(() -> cache.close());
+        }
+      }
 
-      CommandResultAssert replicateRegionCommand =
-          executeCommandAndVerifyStatusIsError(gfsh, command);
-      replicateRegionCommand.hasTableSection(ResultModel.MEMBER_STATUS_SECTION).hasColumn("Message")
-          .asList()
-          .allMatch(s -> s.startsWith(CliStrings.format(REPLICATE_REGION__MSG__EXECUTION__FAILED,
-              "")));
-    }
+      CommandResultAssert result = replicateCommandFuture.get();
+      System.out.println(
+          "Command result after stopping receiver: \n" + result.getCommandResult().asString());
+      // Verify result
+      if (useParallel) {
+        verifyResultOfStoppingReceiverWhenUsingParallelSender(result);
+      } else {
+        verifyResultOfStoppingReceiverWhenUsingSerialSender(result);
+        server2InB.invoke(() -> {
+          assertThat(cache.getRegion(SEPARATOR + regionName).keySet().size()).isEqualTo(entries);
```

Comment:
Any reason for not using validateRegion as in other checks?

Reply:
The check was different when I first added it there. I finally set it to "isEqualTo" and forgot about "validateRegion" function. I will change it.